### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by bolkobe**
I modified the calculation of planetary orbits for the webgui.
In the previous version only Keplers third law of planetary moition was implemented correctly (compare https://en.wikipedia.org/wiki/Kepler%27s_laws_of_planetary_motion).
I implemented the correct ellipse parametrisation as well as a numeric solution to keplers equation (https://en.wikipedia.org/wiki/Kepler%27s_equation), which basically enforces that the planet moves faster if close to the star. So now the orbits (still in 2d instead of 3d) are displayed correctly.

In addition I experimented with a time_scaling factor to change the time step size. Maybe someone with more experience in javascript can put control elements for this on the gui or even implements an automatic adjustment based on the planetary system (average orbital periode or something???).

A last remark for now Asteroid belts should not be displayed as a single orbit since the contain many objects with different orbit orientation (rotationValue in planets.js). In order to avoid plotting a large number of orbits two circles, with radii corresponding to min and max radius, should be drawn maybe with a shaded area in between.

Sorry for intruding in the project without advance notice. Please check my changes and consider a merge.

----
fixed the display of planetary orbits

in the previous version planetary orbits where displayed wrongly -- Periapsis and apoapsis got confused with the semi-major and semi-minor axis as well as the star was set to the center instead in one of the foci of the ellipse (first kepler law)

planetary movement now respects the second kepler law by numeric calculation of kepler equation

time scaling factor is included in gui, this should either be adjusted automatically or controllable by gui elements

### CODE REVIEW

**Comment by MyrionPhoenixmoon** [comment]
First off, welcome! It's great to get contributions from others! :smile: 
Thank you for your detailed explanation of what you did, that is very helpful. 

I have no idea of astronomy, so the implementation of the orbits was simple and your input is appreciated.
I will review and locally test the code tonight and then likely merge it. 

May I also ask how you found the Starsystem generator?

**Comment by bolkobe** [comment]
My idea concerning the time scaling factor is to control the speed of the animation, the year does not change just the dial will complete slower since the change in line 31.
A major issue or motivation is that the orbital period in the inner system is much higher than in the outer system. So either one goes too fast or slow -- depending on the focus of the user it would be good to have a good scale for the orbit one is interested in.

I found the project some time ago, I was working (and still are to some extend on and off) on a code to calculate trajectories in a star system since most tables in RPG rule books concerning travel times are based on assumptions which are in general not fulfilled. My early test where done in our solar system but I needed some star system generator to test it in a more general context.
Basically I am looking into your project and off-course to accrete-starform-stargen.

**Comment by MyrionPhoenixmoon** [comment]
Thank you for implementing my requests!

Hmm. Perhaps you could look at GURPS Space and Spaceships. They have some extensive info on such trajectories. The GURPS forums might also be helpful (found at forums.sjgames.com).

It the meantime, I'll wish you luck and merge your changes!
